### PR TITLE
replace multiple relevancy check queries per domain with single one

### PR DESCRIPTION
**Pull request by sravfeyn**
@pr33thi 

This updates the logic to do a single ES query filtering relevant domains at once instead of, get all domains and check relevancy for each domain using another ES query. In hopes of avoiding https://sentry.io/organizations/dimagi/issues/421161538/events/?project=136860&statsPeriod=14d&utc=false

### CODE REVIEW

File: corehq/apps/reports/tasks.py [hunk 1]
```diff
@@ -168,23 +170,25 @@ def update_calculated_properties():
...
+    Returns ES filter to filter domains that are never updated or
+        domains that haven't been updated since a week or domains that
+        have been updated within last week but have new form submissions
+        in the last day.
+    """
+    last_week = datetime.utcnow() - timedelta(days=7)
+    more_than_a_week_ago = filters.date_range('cp_last_updated', lt=last_week)
+    less_than_a_week_ago = filters.date_range('cp_last_updated', gte=last_week)
+    not_updated = filters.missing('cp_last_updated')
+    domains_submitted_today = filters.term('name',
```
Comment (snopoke) [inline corehq/apps/reports/tasks.py]: how may domains is this likely to be?

**Review by snopoke — APPROVED** [review]

File: corehq/apps/reports/tasks.py (hunk 1 above)
Comment (sravfeyn) [reply corehq/apps/reports/tasks.py]: These were around 400 on prod yesterday when I checked.

Comment (sravfeyn) [reply corehq/apps/reports/tasks.py]: Though, it would be nice to pass `distinct` to the ES query itself, but I haven't found an inbuilt filter for it.

Comment (sravfeyn) [reply corehq/apps/reports/tasks.py]: Ha, found it https://github.com/dimagi/commcare-hq/pull/23727/commits/d4ffb6577b86eda2e59e68702973f078b20c4248

File: corehq/apps/reports/tasks.py [hunk 2]
```diff
@@ -168,23 +171,29 @@ def update_calculated_properties():
...
+    more_than_a_week_ago = filters.date_range('cp_last_updated', lt=last_week)
+    less_than_a_week_ago = filters.date_range('cp_last_updated', gte=last_week)
+    not_updated = filters.missing('cp_last_updated')
+    domains_submitted_today = [
+        agg.domain
+        for agg in NestedTermAggregationsHelper(
+            base_query=FormES().submitted(gte=datetime.utcnow() - timedelta(days=1)),
+            terms=[AggregationTerm('domain', 'domain')]
+        ).get_data()
+    ]
```
Comment (snopoke) [inline corehq/apps/reports/tasks.py]: you can just use a normal terms aggregation:
```python
FormES().submitted(gte=datetime.utcnow() - timedelta(days=1))
    .terms_aggregation('domain', 'domain')
    .size(0)
    .run()
    .aggregations.domain.keys
```

**Review by kaapstorm — APPROVED** [review]
Nice

File: corehq/apps/reports/tasks.py [hunk 3]
```diff
@@ -83,23 +84,24 @@ def update_calculated_properties():
...
+    last_week = datetime.utcnow() - timedelta(days=7)
+    more_than_a_week_ago = filters.date_range('cp_last_updated', lt=last_week)
+    less_than_a_week_ago = filters.date_range('cp_last_updated', gte=last_week)
+    not_updated = filters.missing('cp_last_updated')
+    domains_submitted_today = (FormES().submitted(gte=datetime.utcnow() - timedelta(days=1))
+        .terms_aggregation('domain', 'domain').size(0).run().aggregations.domain.keys)
+    return filters.OR(
+        not_updated,
+        more_than_a_week_ago,
+        filters.AND(less_than_a_week_ago, domains_submitted_today)
```
Comment (snopoke) [inline corehq/apps/reports/tasks.py]: @sravfeyn I don't think this is correct. `domains_submitted_today` is just a list of domain names and not a filter. It should be:
```python
filters.AND(less_than_a_week_ago, filters.term('name', domains_submitted_today))
```

Comment (sravfeyn) [reply corehq/apps/reports/tasks.py]: Thanks https://github.com/dimagi/commcare-hq/pull/23835

**Pull request merged**